### PR TITLE
fix for #57

### PR DESCRIPTION
**Pull request by daniel-ciaglia**
### Checklist

- [x] All new jobs, commands, executors, parameters have descriptions
- [x] Examples have been added for any significant new features
- [x] README has been updated, if necessary

### Motivation, issues
Fixes issue #57

### Description
added support for `source-profile`
do not set credentials on `role-arn` profiles

### CODE REVIEW

File: src/commands/setup.yml [hunk 1]
```diff
@@ -56,7 +56,12 @@ parameters:
     default: false
 
   role-arn:
-    description: Role ARN that the profile should take.
+    description: Role ARN that the profile should take. (needs source_profile)
+    type: string
+    default: ""
+
```
Comment (gmemstr) [inline src/commands/setup.yml:62]: Linting

**Review by gmemstr — CHANGES_REQUESTED** [review]
Minor linting fixes

**Comment by daniel-ciaglia** [comment]
question - can I test the code somehow privately?

File: .circleci/config.yml [hunk 2]
```diff
@@ -39,7 +39,8 @@ integration_tests: &integration_tests
     # v1-docker,
     integration-test-skip-install,
     integration-test-install-v1,
-    replace-pip-aws-test
+    replace-pip-aws-test,
+    use-role-arn
```
Comment (gmemstr) [inline .circleci/config.yml:43]: Has to be added to the relevant workflow.

**Review by gmemstr — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 2 above)
Comment (daniel-ciaglia) [reply .circleci/config.yml:43]: should be in.

**Comment by KyleTryon** [comment]
Hello @daniel-ciaglia Thank you for your contribution. If you don't mind, I would like to hear more on your motivations for this change. I was not actually aware of the role-arn parameter added to this orb but I am doing some research in this area currently.

My thinking was, you would want to assume a role, to prevent having any static credentials. Is providing the static account in this process defeating that purpose of assuming a role?

**Comment by daniel-ciaglia** [comment]
Hi @KyleTryon, I described the the motivation in the underlying issue #57.
Would you mind checking this and see if this is enough? Cheers, Daniel

File: README.md [hunk 3]
```diff
@@ -32,3 +32,40 @@ workflows:
...
+orbs:
+  aws-cli: circleci/aws-cli@x.y
+
+jobs:
+  aws-cli-example:
+    executor: aws-cli/default
+    steps:
+      - checkout
+      - aws-cli/setup  # set up the `default` profile with static credentials
+      - aws-cli/setup: # set up the profile with to be assumed role
```
Comment (KyleTryon) [inline README.md:56]: Could we move this from the readme to a new usage example?

File: src/commands/setup.yml [hunk 4]
```diff
@@ -56,7 +56,12 @@ parameters:
...
+  source-profile:
```
Comment (AlanRaisonNews) [inline src/commands/setup.yml:63]: Rather than requiring this to be set, could it be calculated, e.g.
```
<<parameters.role-arn>>-source
```

**Comment by brivu** [comment]
Hi All, 

Support for configuring `role-arn` has been added in the next major release: #98, 

Thanks for your contributions!

Best, 
Brian

**Pull request closed without merging**